adapters/InMobi/InMobiAdapter/GADMAdapterInMobiInitializer.m

```diff
@@ -61,28 +61,43 @@ - (void)initializeWithAccountID:(nonnull NSString *)accountID
   }
 
   _initializationState = GADMAdapterInMobiInitStateInitializing;
-  GADMAdapterInMobiInitializer *__weak weakSelf = self;
-  [IMSdk initWithAccountID:accountID
-         consentDictionary:GADMInMobiConsent.consent
-      andCompletionHandler:^(NSError *_Nullable error) {
-        GADMAdapterInMobiInitializer *strongSelf = weakSelf;
-        if (!strongSelf) {
-          return;
-        }
-
-        if (error) {
-          strongSelf->_initializationState = GADMAdapterInMobiInitStateUninitialized;
-        } else {
-          GADMAdapterInMobiLog(@"InMobi SDK Initialized successfully.");
-
-          strongSelf->_initializationState = GADMAdapterInMobiInitStateInitialized;
-        }
-
-        for (GADMAdapterInMobiInitCompletionHandler completionHandler in strongSelf
-                 ->_completionHandlers) {
-          completionHandler(error);
-        }
-      }];
+    GADMAdapterInMobiInitializer *__weak weakSelf = self;
+
+    if ([NSThread isMainThread]) {
+        [self initializeInMobiSDK:accountID completionHandler:completionHandler];
+    } else {
```

Comment:
I think this check is not necessary. 
Calling `dispatch_async(dispatch_get_main_queue() ...` should already add this to the queue on the main thread, even if its already in the main thread.

Reply:
without this check init call would become async if already on main thread. 
We want ii to remain sync if already on main thread